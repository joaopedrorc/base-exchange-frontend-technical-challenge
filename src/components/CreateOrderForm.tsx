'use client';

import { z } from 'zod';
import { zodResolver } from '@hookform/resolvers/zod';
import { useForm } from 'react-hook-form';
import {
  Form,
  FormControl,
  FormDescription,
  FormField,
  FormItem,
  FormLabel,
  FormMessage,
} from '@/components/ui/form';
import {
  Select,
  SelectContent,
  SelectGroup,
  SelectItem,
  SelectTrigger,
  SelectValue,
} from '@/components/ui/select';
import { Input } from '@/components/ui/input';
import { Button } from '@/components/ui/button';
import { useState } from 'react';

const formSchema = z.object({
  asset: z.string().min(1, 'Ativo é obrigatório'),
  side: z.enum(['Compra', 'Venda'], {
    required_error: 'Selecione uma opção',
  }),
  price: z
    .string()
    .regex(
      /^(\s?R\$)?\s?\d{1,3}(\.\d{3})*(,\d{2})?$/,
      'Preço inválido, use o formato R$ 1.000,00'
    )
    .min(1, 'Preço é obrigatório'),
  amount: z
    .string()
    .min(1, { message: 'Mínimo de 1 unidade' })
    .max(1000, { message: 'Máximo de 1000 unidades' }),
});

export function CreateOrderForm() {
  const [loading, setLoading] = useState(false);
  const form = useForm<z.infer<typeof formSchema>>({
    resolver: zodResolver(formSchema),
    defaultValues: {
      asset: '',
      side: 'Compra',
      price: '',
      amount: '',
    },
  });

  async function onSubmit(values: z.infer<typeof formSchema>) {
    try {
      setLoading(true);
      const res = await fetch('/api/orders', {
        method: 'POST',
        body: JSON.stringify(values),
      });

      const data = await res.json();
      console.log(data);
      setLoading(false);
    } catch (error) {
      console.log('ERROR', error);

      setLoading(false);
    }

    console.log(values);
  }

  function formatToBRL(value: string) {
    const numericValue = value.replace(/\D/g, '');
    const amount = parseInt(numericValue || '0', 10) / 100;
    return amount.toLocaleString('pt-BR', {
      style: 'currency',
      currency: 'BRL',
    });
  }

  function formatInteger(value: string) {
    let numericValue = value.replace(/\D/g, '');
    let intValue = parseInt(numericValue || '0', 10);

    if (intValue > 1000) {
      intValue = 1000;
    }

    return intValue > 0 ? intValue.toString() : '';
  }

  return (
    <div>
      <Form {...form}>
        <form
          onSubmit={form.handleSubmit(onSubmit)}
          className="grid grid-cols-1 gap-4 space-y-8 lg:grid-cols-4"
        >
          <FormField
            control={form.control}
            name="asset"
            render={({ field }) => (
              <FormItem>
                <FormLabel>Ações</FormLabel>
                <FormControl>
                  <Select
                    onValueChange={field.onChange}
                    defaultValue={field.value}
                  >
                    <SelectTrigger className="w-full">
                      <SelectValue placeholder="Selecione uma ação" />
                    </SelectTrigger>
                    <SelectContent>
                      <SelectGroup>
                        <SelectItem value="B3SA3">B3 (B3SA3)</SelectItem>
                        <SelectItem value="BBAS3">
                          Banco do Brasil (BBAS3)
                        </SelectItem>
                        <SelectItem value="PETR4">Petrobras (PETR4)</SelectItem>
                        <SelectItem value="HAPV3">Hapvida (HAPV3)</SelectItem>
                        <SelectItem value="ITUB4">
                          Itaú Unibanco (ITUB4)
                        </SelectItem>
                        <SelectItem value="BBDC4">Bradesco (BBDC4)</SelectItem>
                        <SelectItem value="VALE3">Vale (VALE3)</SelectItem>
                        <SelectItem value="CMIG4">Cemig (CMIG4)</SelectItem>
                        <SelectItem value="ABEV3">Ambev (ABEV3)</SelectItem>
                        <SelectItem value="BEEF3">Minerva (BEEF3)</SelectItem>
                        <SelectItem value="MGLU3">
                          Magazine Luiza (MGLU3)
                        </SelectItem>
                        <SelectItem value="PETZ3">
                          Pet Center (PETZ3)
                        </SelectItem>
                        <SelectItem value="RADL3">
                          Raia Drogasil (RADL3)
                        </SelectItem>
                        <SelectItem value="WEGE3">WEG (WEGE3)</SelectItem>
                        <SelectItem value="SUZB3">Suzano (SUZB3)</SelectItem>
                        <SelectItem value="RENT3">Localiza (RENT3)</SelectItem>
                        <SelectItem value="ELET3">
                          Eletrobras (ELET3)
                        </SelectItem>
                        <SelectItem value="GGBR4">Gerdau (GGBR4)</SelectItem>
                        <SelectItem value="CSNA3">CSN (CSNA3)</SelectItem>
                        <SelectItem value="KLBN11">Klabin (KLBN11)​</SelectItem>
                      </SelectGroup>
                    </SelectContent>
                  </Select>
                </FormControl>
                <FormDescription>As 20 ações mais negociadas.</FormDescription>
                <FormMessage />
              </FormItem>
            )}
          />
          <FormField
            control={form.control}
            name="side"
            render={({ field }) => (
              <FormItem>
                <FormLabel>Operação</FormLabel>
                <FormControl>
                  <Select
                    onValueChange={field.onChange}
                    defaultValue={field.value}
                  >
                    <SelectTrigger className="w-full">
                      <SelectValue placeholder="Selecione uma operação" />
                    </SelectTrigger>
                    <SelectContent>
                      <SelectGroup>
                        <SelectItem value="Compra">Compra</SelectItem>
                        <SelectItem value="Venda">Venda</SelectItem>
                      </SelectGroup>
                    </SelectContent>
                  </Select>
                </FormControl>
                <FormDescription>Operação de compra/venda.</FormDescription>
                <FormMessage />
              </FormItem>
            )}
          />
          <FormField
            control={form.control}
            name="price"
            render={({ field }) => (
              <FormItem>
                <FormLabel>Preço</FormLabel>
                <FormControl>
                  <Input
                    {...field}
                    type="text"
                    inputMode="numeric"
                    pattern="^(\s?R\$)?\s?\d{1,3}(\.\d{3})*(,\d{2})?$"
                    placeholder="R$ 10,99"
                    onChange={(e) =>
                      field.onChange(formatToBRL(e.target.value))
                    }
                    value={field.value}
                  />
                </FormControl>
                <FormDescription>Preço de negociação.</FormDescription>
                <FormMessage />
              </FormItem>
            )}
          />
          <FormField
            control={form.control}
            name="amount"
            render={({ field }) => (
              <FormItem>
                <FormLabel>Quantidade</FormLabel>
                <FormControl>
                  <Input
                    {...field}
                    value={field.value}
                    onChange={(e) =>
                      field.onChange(formatInteger(e.target.value))
                    }
                    type="text"
                    inputMode="numeric"
                    placeholder="2 un."
                  />
                </FormControl>
                <FormDescription>Digite um número de 1 a 1000.</FormDescription>
                <FormMessage />
              </FormItem>
            )}
          />
          <Button type="submit" disabled={loading}>
            {loading ? 'enviando' : 'Criar ordem'}
          </Button>
        </form>
      </Form>
    </div>
  );
}
