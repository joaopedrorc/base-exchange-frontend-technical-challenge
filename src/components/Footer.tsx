export function Footer() {
  return (
    <footer className="mt-auto w-full bg-slate-800 px-4 py-3 text-white">
      <div className="mx-auto flex max-w-7xl flex-col items-center justify-center text-sm sm:flex-row">
        <span>&copy; {new Date().getFullYear()} Gerenciador de Ordens</span>
      </div>
    </footer>
  );
}
