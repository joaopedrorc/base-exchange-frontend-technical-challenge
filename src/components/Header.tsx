import { Button } from '@/components/ui/button';
import { cn } from '@/lib/utils';

export function Header() {
  return (
    <header className="w-full bg-slate-800 px-4 py-3 text-white shadow-md">
      <div className="mx-auto flex max-w-7xl items-center justify-between">
        {/* Left: App Title */}
        <h1 className="text-xl font-semibold">Gerenciador de Ordens</h1>

        {/* Right: Buttons */}
        <div className="flex gap-2">
          <Button
            variant="secondary"
            className={cn('bg-slate-700 text-white hover:bg-slate-600')}
          >
            Home
          </Button>
          <Button className="bg-white text-slate-800 hover:bg-slate-100">
            Nova ordem
          </Button>
        </div>
      </div>
    </header>
  );
}
