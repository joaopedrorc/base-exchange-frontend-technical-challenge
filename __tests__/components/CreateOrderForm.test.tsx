import { render } from '@testing-library/react';
import { CreateOrderForm } from '@/components/CreateOrderForm';

describe('CreateOrderForm component', () => {
  it('should render CreateOrderForm component', () => {
    render(<CreateOrderForm />);
  });

  it('renders CreateOrderForm component unchanged', async () => {
    const component = render(<CreateOrderForm />);

    expect(component).toMatchSnapshot();
  });
});
