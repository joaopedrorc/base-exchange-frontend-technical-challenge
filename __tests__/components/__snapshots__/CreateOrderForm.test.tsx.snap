// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateOrderForm component renders CreateOrderForm component unchanged 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <form
          class="grid grid-cols-1 gap-4 space-y-8 lg:grid-cols-4"
        >
          <div
            class="grid gap-2"
            data-slot="form-item"
          >
            <label
              class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
              data-error="false"
              data-slot="form-label"
              for="«ru»-form-item"
            >
              Ações
            </label>
            <button
              aria-autocomplete="none"
              aria-controls="radix-«rv»"
              aria-expanded="false"
              class="border-input data-[placeholder]:text-muted-foreground [&_svg:not([class*='text-'])]:text-muted-foreground focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive dark:bg-input/30 dark:hover:bg-input/50 flex items-center justify-between gap-2 rounded-md border bg-transparent px-3 py-2 text-sm whitespace-nowrap shadow-xs transition-[color,box-shadow] outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50 data-[size=default]:h-9 data-[size=sm]:h-8 *:data-[slot=select-value]:line-clamp-1 *:data-[slot=select-value]:flex *:data-[slot=select-value]:items-center *:data-[slot=select-value]:gap-2 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4 w-full"
              data-placeholder=""
              data-size="default"
              data-slot="select-trigger"
              data-state="closed"
              dir="ltr"
              role="combobox"
              type="button"
            >
              <span
                data-slot="select-value"
                style="pointer-events: none;"
              >
                Selecione uma ação
              </span>
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-down size-4 opacity-50"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
            <select
              aria-hidden="true"
              style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
              tabindex="-1"
            >
              <option
                value="B3SA3"
              >
                B3 (B3SA3)
              </option>
              <option
                value="BBAS3"
              >
                Banco do Brasil (BBAS3)
              </option>
              <option
                value="PETR4"
              >
                Petrobras (PETR4)
              </option>
              <option
                value="HAPV3"
              >
                Hapvida (HAPV3)
              </option>
              <option
                value="ITUB4"
              >
                Itaú Unibanco (ITUB4)
              </option>
              <option
                value="BBDC4"
              >
                Bradesco (BBDC4)
              </option>
              <option
                value="VALE3"
              >
                Vale (VALE3)
              </option>
              <option
                value="CMIG4"
              >
                Cemig (CMIG4)
              </option>
              <option
                value="ABEV3"
              >
                Ambev (ABEV3)
              </option>
              <option
                value="BEEF3"
              >
                Minerva (BEEF3)
              </option>
              <option
                value="MGLU3"
              >
                Magazine Luiza (MGLU3)
              </option>
              <option
                value="PETZ3"
              >
                Pet Center (PETZ3)
              </option>
              <option
                value="RADL3"
              >
                Raia Drogasil (RADL3)
              </option>
              <option
                value="WEGE3"
              >
                WEG (WEGE3)
              </option>
              <option
                value="SUZB3"
              >
                Suzano (SUZB3)
              </option>
              <option
                value="RENT3"
              >
                Localiza (RENT3)
              </option>
              <option
                value="ELET3"
              >
                Eletrobras (ELET3)
              </option>
              <option
                value="GGBR4"
              >
                Gerdau (GGBR4)
              </option>
              <option
                value="CSNA3"
              >
                CSN (CSNA3)
              </option>
              <option
                value="KLBN11"
              >
                Klabin (KLBN11)​
              </option>
            </select>
            <p
              class="text-muted-foreground text-sm"
              data-slot="form-description"
              id="«ru»-form-item-description"
            >
              As 20 ações mais negociadas.
            </p>
          </div>
          <div
            class="grid gap-2"
            data-slot="form-item"
          >
            <label
              class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
              data-error="false"
              data-slot="form-label"
              for="«r10»-form-item"
            >
              Operação
            </label>
            <button
              aria-autocomplete="none"
              aria-controls="radix-«r11»"
              aria-expanded="false"
              class="border-input data-[placeholder]:text-muted-foreground [&_svg:not([class*='text-'])]:text-muted-foreground focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive dark:bg-input/30 dark:hover:bg-input/50 flex items-center justify-between gap-2 rounded-md border bg-transparent px-3 py-2 text-sm whitespace-nowrap shadow-xs transition-[color,box-shadow] outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50 data-[size=default]:h-9 data-[size=sm]:h-8 *:data-[slot=select-value]:line-clamp-1 *:data-[slot=select-value]:flex *:data-[slot=select-value]:items-center *:data-[slot=select-value]:gap-2 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4 w-full"
              data-size="default"
              data-slot="select-trigger"
              data-state="closed"
              dir="ltr"
              role="combobox"
              type="button"
            >
              <span
                data-slot="select-value"
                style="pointer-events: none;"
              >
                Compra
              </span>
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-down size-4 opacity-50"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
            <select
              aria-hidden="true"
              style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
              tabindex="-1"
            >
              <option
                selected=""
                value="Compra"
              >
                Compra
              </option>
              <option
                value="Venda"
              >
                Venda
              </option>
            </select>
            <p
              class="text-muted-foreground text-sm"
              data-slot="form-description"
              id="«r10»-form-item-description"
            >
              Operação de compra/venda.
            </p>
          </div>
          <div
            class="grid gap-2"
            data-slot="form-item"
          >
            <label
              class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
              data-error="false"
              data-slot="form-label"
              for="«r12»-form-item"
            >
              Preço
            </label>
            <input
              aria-describedby="«r12»-form-item-description"
              aria-invalid="false"
              class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
              data-slot="form-control"
              id="«r12»-form-item"
              inputmode="numeric"
              name="price"
              pattern="^(\\s?R\\$)?\\s?\\d{1,3}(\\.\\d{3})*(,\\d{2})?$"
              placeholder="R$ 10,99"
              type="text"
              value=""
            />
            <p
              class="text-muted-foreground text-sm"
              data-slot="form-description"
              id="«r12»-form-item-description"
            >
              Preço de negociação.
            </p>
          </div>
          <div
            class="grid gap-2"
            data-slot="form-item"
          >
            <label
              class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
              data-error="false"
              data-slot="form-label"
              for="«r13»-form-item"
            >
              Quantidade
            </label>
            <input
              aria-describedby="«r13»-form-item-description"
              aria-invalid="false"
              class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
              data-slot="form-control"
              id="«r13»-form-item"
              inputmode="numeric"
              name="amount"
              placeholder="2 un."
              type="text"
              value=""
            />
            <p
              class="text-muted-foreground text-sm"
              data-slot="form-description"
              id="«r13»-form-item-description"
            >
              Digite um número de 1 a 1000.
            </p>
          </div>
          <button
            class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3"
            data-slot="button"
            type="submit"
          >
            Criar ordem
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <form
        class="grid grid-cols-1 gap-4 space-y-8 lg:grid-cols-4"
      >
        <div
          class="grid gap-2"
          data-slot="form-item"
        >
          <label
            class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
            data-error="false"
            data-slot="form-label"
            for="«ru»-form-item"
          >
            Ações
          </label>
          <button
            aria-autocomplete="none"
            aria-controls="radix-«rv»"
            aria-expanded="false"
            class="border-input data-[placeholder]:text-muted-foreground [&_svg:not([class*='text-'])]:text-muted-foreground focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive dark:bg-input/30 dark:hover:bg-input/50 flex items-center justify-between gap-2 rounded-md border bg-transparent px-3 py-2 text-sm whitespace-nowrap shadow-xs transition-[color,box-shadow] outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50 data-[size=default]:h-9 data-[size=sm]:h-8 *:data-[slot=select-value]:line-clamp-1 *:data-[slot=select-value]:flex *:data-[slot=select-value]:items-center *:data-[slot=select-value]:gap-2 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4 w-full"
            data-placeholder=""
            data-size="default"
            data-slot="select-trigger"
            data-state="closed"
            dir="ltr"
            role="combobox"
            type="button"
          >
            <span
              data-slot="select-value"
              style="pointer-events: none;"
            >
              Selecione uma ação
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-down size-4 opacity-50"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m6 9 6 6 6-6"
              />
            </svg>
          </button>
          <select
            aria-hidden="true"
            style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
            tabindex="-1"
          >
            <option
              value="B3SA3"
            >
              B3 (B3SA3)
            </option>
            <option
              value="BBAS3"
            >
              Banco do Brasil (BBAS3)
            </option>
            <option
              value="PETR4"
            >
              Petrobras (PETR4)
            </option>
            <option
              value="HAPV3"
            >
              Hapvida (HAPV3)
            </option>
            <option
              value="ITUB4"
            >
              Itaú Unibanco (ITUB4)
            </option>
            <option
              value="BBDC4"
            >
              Bradesco (BBDC4)
            </option>
            <option
              value="VALE3"
            >
              Vale (VALE3)
            </option>
            <option
              value="CMIG4"
            >
              Cemig (CMIG4)
            </option>
            <option
              value="ABEV3"
            >
              Ambev (ABEV3)
            </option>
            <option
              value="BEEF3"
            >
              Minerva (BEEF3)
            </option>
            <option
              value="MGLU3"
            >
              Magazine Luiza (MGLU3)
            </option>
            <option
              value="PETZ3"
            >
              Pet Center (PETZ3)
            </option>
            <option
              value="RADL3"
            >
              Raia Drogasil (RADL3)
            </option>
            <option
              value="WEGE3"
            >
              WEG (WEGE3)
            </option>
            <option
              value="SUZB3"
            >
              Suzano (SUZB3)
            </option>
            <option
              value="RENT3"
            >
              Localiza (RENT3)
            </option>
            <option
              value="ELET3"
            >
              Eletrobras (ELET3)
            </option>
            <option
              value="GGBR4"
            >
              Gerdau (GGBR4)
            </option>
            <option
              value="CSNA3"
            >
              CSN (CSNA3)
            </option>
            <option
              value="KLBN11"
            >
              Klabin (KLBN11)​
            </option>
          </select>
          <p
            class="text-muted-foreground text-sm"
            data-slot="form-description"
            id="«ru»-form-item-description"
          >
            As 20 ações mais negociadas.
          </p>
        </div>
        <div
          class="grid gap-2"
          data-slot="form-item"
        >
          <label
            class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
            data-error="false"
            data-slot="form-label"
            for="«r10»-form-item"
          >
            Operação
          </label>
          <button
            aria-autocomplete="none"
            aria-controls="radix-«r11»"
            aria-expanded="false"
            class="border-input data-[placeholder]:text-muted-foreground [&_svg:not([class*='text-'])]:text-muted-foreground focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive dark:bg-input/30 dark:hover:bg-input/50 flex items-center justify-between gap-2 rounded-md border bg-transparent px-3 py-2 text-sm whitespace-nowrap shadow-xs transition-[color,box-shadow] outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50 data-[size=default]:h-9 data-[size=sm]:h-8 *:data-[slot=select-value]:line-clamp-1 *:data-[slot=select-value]:flex *:data-[slot=select-value]:items-center *:data-[slot=select-value]:gap-2 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4 w-full"
            data-size="default"
            data-slot="select-trigger"
            data-state="closed"
            dir="ltr"
            role="combobox"
            type="button"
          >
            <span
              data-slot="select-value"
              style="pointer-events: none;"
            >
              Compra
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-down size-4 opacity-50"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m6 9 6 6 6-6"
              />
            </svg>
          </button>
          <select
            aria-hidden="true"
            style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
            tabindex="-1"
          >
            <option
              selected=""
              value="Compra"
            >
              Compra
            </option>
            <option
              value="Venda"
            >
              Venda
            </option>
          </select>
          <p
            class="text-muted-foreground text-sm"
            data-slot="form-description"
            id="«r10»-form-item-description"
          >
            Operação de compra/venda.
          </p>
        </div>
        <div
          class="grid gap-2"
          data-slot="form-item"
        >
          <label
            class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
            data-error="false"
            data-slot="form-label"
            for="«r12»-form-item"
          >
            Preço
          </label>
          <input
            aria-describedby="«r12»-form-item-description"
            aria-invalid="false"
            class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
            data-slot="form-control"
            id="«r12»-form-item"
            inputmode="numeric"
            name="price"
            pattern="^(\\s?R\\$)?\\s?\\d{1,3}(\\.\\d{3})*(,\\d{2})?$"
            placeholder="R$ 10,99"
            type="text"
            value=""
          />
          <p
            class="text-muted-foreground text-sm"
            data-slot="form-description"
            id="«r12»-form-item-description"
          >
            Preço de negociação.
          </p>
        </div>
        <div
          class="grid gap-2"
          data-slot="form-item"
        >
          <label
            class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
            data-error="false"
            data-slot="form-label"
            for="«r13»-form-item"
          >
            Quantidade
          </label>
          <input
            aria-describedby="«r13»-form-item-description"
            aria-invalid="false"
            class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
            data-slot="form-control"
            id="«r13»-form-item"
            inputmode="numeric"
            name="amount"
            placeholder="2 un."
            type="text"
            value=""
          />
          <p
            class="text-muted-foreground text-sm"
            data-slot="form-description"
            id="«r13»-form-item-description"
          >
            Digite um número de 1 a 1000.
          </p>
        </div>
        <button
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3"
          data-slot="button"
          type="submit"
        >
          Criar ordem
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
